.github/workflows/main.yml

```diff
@@ -28,5 +33,48 @@ jobs:
         with:
           java-version: ${{matrix.jdk}}
           distribution: 'temurin'
-      - name: Build and test
-        run: ./.ci-build.sh
+      - name: Build
+        id: build
+        run: ./.ci-build-without-test.sh
+
+      - name: PICO Typecheck Test
```

Comment:
Aren't all of these steps in the same job?
Wouldn't it be nicer to have different jobs for the different targets, to not stop everything as soon as there is a failure?
Maybe above under `matrix` you can add a `target: [ImmutabilityTypecheckTest, ImmutabilityTypecheckBaseAllSystemsTest, ImmutabilityTypecheckGlacierTest...]` and then they run `./gradlew ${{ matrix.target }}` or something?